api-operator/pkg/mgw/deployment.go

```diff
@@ -62,11 +84,36 @@ func AddContainers(containers *[]corev1.Container) {
 }
 
 // Deployment returns a MGW deployment for the given API definition
-func Deployment(api *wso2v1alpha1.API, controlConfigData map[string]string, owner *[]metav1.OwnerReference) *appsv1.Deployment {
+func Deployment(client *client.Client, api *wso2v1alpha1.API, controlConfigData map[string]string,
+	owner *[]metav1.OwnerReference) *appsv1.Deployment {
 	regConfig := registry.GetConfig()
 	labels := map[string]string{"app": api.Name}
 	var deployVolume []corev1.Volume
 	var deployVolumeMount []corev1.VolumeMount
+	var mgwDeployVol *v1.Volume
+	var mgwDeployMount *v1.VolumeMount
+	mgwDeploymentConfMap := k8s.NewConfMap()
+	errGetDeploy := k8s.Get(client, types.NamespacedName{Name: mgwDeploymentConfigMapName, Namespace: api.Namespace}, mgwDeploymentConfMap)
+	if errGetDeploy != nil && errors.IsNotFound(errGetDeploy) {
+		logDeploy.Info("Get mgw deployment configs", "from namespace", wso2NameSpaceConst)
+		//retrieve mgw deployment configs from wso2-system namespace
+		err := k8s.Get(client, types.NamespacedName{Namespace: wso2NameSpaceConst, Name: mgwDeploymentConfigMapName},
+			mgwDeploymentConfMap)
+		if err != nil {
+			logDeploy.Error(err, "MGW Deployment configs not defined")
+			return nil
```

Comment:
Shouldn't we return error if it is not a `errors.IsNotFound(err)`? if it is not found can't we continue without returning.
```suggestion
			if !errors.IsNotFound(err){
			    return err
			}
			// else continue without returning
```

Reply:
Fixed with 533f5d267babd0d26a517e9eaff29e76d9a34e91